includes/discovery/discovery-protocols.inc.php

```diff
@@ -126,7 +126,7 @@
             if (is_numeric($dot1d_array[$entry_key]['dot1dBasePortIfIndex'])) {
                 $ifIndex = $dot1d_array[$entry_key]['dot1dBasePortIfIndex'];
             } else {
-                $ifIndex = $entry_key;
+                $ifIndex = null;
```

Comment:
This will break other devices.

Reply:
This sets an ifIndex = 1, 2, 3, 4, 5..... 

According to my knowledge, the physical ports are not at that index in SNMP. At least for my devices, the only things at those indices are VLAN1, VLAN2, VLAN3, VLAN4..... 


Also... as I mentioned in commit.... the ifIndex is only used for local_port finding. The LLDP should give the interface name or mac or something better. 


I do not know many devices, so this might be working. But this seems like a hack to assume that if the ifIndex is not dot1dbaseportifindex, it should be set to the index of the array.

Reply:
Setting it to null might be better, because then it ignores arbritary lookups.

Reply:
@murrant https://github.com/librenms/librenms/pull/7564 actually solves my problems, So I am ok if you don't want to apply this.